projects/cm_mcu/MonitorI2CTask.c

```diff
@@ -98,8 +101,33 @@ void MonitorI2CTask(void *parameters)
       log_debug(LOG_MONI2C, "%s: device %d\r\n", args->name, ps);
 
       if (ps == args->n_devices - 1 && getPowerControlState() != POWER_ON) { // avoid continues to infinite loops due to multi-threading when pwr is not on
```

Comment:
Hi @pkotamnives what is the logic behind the first part of the `if` statement (` ps == args->n_devices - 1 `)? It seems like this would allow the `for` loop to attempt readout even if the power is off.